src/snowflake/snowpark/ml/utils.py

```diff
@@ -0,0 +1,140 @@
+    -------------------------------------------------------------------------------------------------------------------------
+    """
+    if len(transformer.input_cols) != len(transformer.output_cols):
+        raise ValueError("The number of input column and output column does not match")
+    df._describe(transformer.input_cols, stats=stats).select(
+        [
+            object_agg("summary", c).as_(fc)
+            for c, fc in zip(transformer.input_cols, transformer._states_table_cols)
+        ]
+    ).write.save_as_table(transformer._states_table_name, create_temp_table=True)
+
+
+def encoder_fit(
+    transformer: "snowflake.snowpark.ml.Transformer", df: DataFrame, encoder_type: str
+):
+    if transformer._category == "auto":
+        df_encoder_length = None
+        for c, fc in zip(transformer.input_cols, transformer._states_table_cols):
+            transformer._category_table_name[c] = random_name_for_temp_object(
+                TempObjectType.TABLE
+            )
+            df_save = df._distinct([c]).select(
+                [
+                    col(c).as_(fc),
+                ]
+            )
+            if encoder_type == "onehot":
+                if df_encoder_length:
+                    df_encoder_length = df_encoder_length.join(
+                        df_save.select([count_distinct(col(fc)).as_(f"{fc}_count")])
+                    )
+                else:
+                    df_encoder_length = df_save.select(
+                        [count_distinct(col(fc)).as_(f"{fc}_count")]
+                    )
+            df_save.with_column(f"{fc}_encoder", builtin("seq8")()).write.save_as_table(
+                transformer._category_table_name[c], create_temp_table=True
+            )
+        if encoder_type == "onehot":
+            temp = lit(0)
+            for c in transformer._states_table_cols:
+                temp += df_encoder_length[f"{c}_count"]
+
+            df_encoder_length.with_column("encoder_count", temp).write.save_as_table(
+                transformer._encoder_count_table,
+                create_temp_table=True,
+                mode="overwrite",
+            )
+    else:
+        session = df._session
+        df_count = {}
+        df_encoder_length = 0
+        for c, fc, cat in zip(
+            transformer.input_cols,
+            transformer._states_table_cols,
+            transformer._category,
+        ):
+            transformer._category_table_name[c] = random_name_for_temp_object(
+                TempObjectType.TABLE
+            )
+            df_save = session.create_dataframe(cat, schema=[fc])
+            if encoder_type == "onehot":
+                df_count[f"{fc}_count"] = len(cat)
+                df_encoder_length += len(cat)
+            df_save.with_column(f"{fc}_encoder", builtin("seq8")()).write.save_as_table(
+                transformer._category_table_name[c], create_temp_table=True
+            )
+        """
```

Comment:
Move up this docstring and the following string? Similar to what you did in scaler_fit?